dd-sdk-android-internal/src/test/java/com/datadog/internal/utils/ByteArrayExtTest.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.internal.utils
+
+import com.datadog.android.internal.utils.toHexString
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+import java.util.Locale
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class ByteArrayExtTest {
```

Comment:
> [!NOTE]
> Since the goal of this is to optimize the performance of converting a bytearray to string, shouldn't we add a performance test ? 
> - Make a large byte array
> - convert it to hex string using this new function, and joinToString
> - compare the performance of both methods

Reply:
Interesting point, However I am not sure the performance test should be part of the unit test, since unit test is supposed to verify **correctness**. Also I am afraid of performance test can be flaky, it depends also on the running environment, the test can fail even though we know in general `toHexString` is better than `joinToString`.

So do you have any suggestion how we should proceed this performance test?

Reply:
I will merge it since I got enough approval, don't hesitate to add anything here, we can do it in another PR!